plugin-server/tests/worker/ingestion/event-pipeline/populateTeamDataStep.test.ts

```diff
@@ -0,0 +1,64 @@
+import { PipelineEvent, Team } from '../../../../src/types'
+import { UUIDT } from '../../../../src/utils/utils'
+import { populateTeamDataStep } from '../../../../src/worker/ingestion/event-pipeline/1-populateTeamDataStep'
+
+const pipelineEvent: PipelineEvent = {
+    event: '$pageview',
+    properties: { foo: 'bar' },
+    timestamp: '2020-02-23T02:15:00Z',
+    now: '2020-02-23T02:15:00Z',
+    team_id: 2,
+    distinct_id: 'my_id',
+    ip: '127.0.0.1',
+    site_url: 'https://example.com',
+    uuid: new UUIDT().toString(),
+    token: 'token',
+}
+
+const { token, ...defaultResultEvent } = pipelineEvent
+
+const team: Team = {
+    id: 2,
+    uuid: 'af95d312-1a0a-4208-b80f-562ddafc9bcd',
+    organization_id: '66f3f7bf-44e2-45dd-9901-5dbd93744e3a',
+    name: 'testTeam',
+    anonymize_ips: false,
+    api_token: 'token',
+    slack_incoming_webhook: '',
+    session_recording_opt_in: false,
+    ingested_event: true,
+}
+
+let runner: any
+
+beforeEach(() => {
+    runner = {
+        nextStep: (...args: any[]) => args,
+        hub: {
+            teamManager: {
+                getTeamByToken: jest.fn(() => team),
```

Comment:
should we be mocking this rather than using the DB in tests? Karl had some point around this, for our team, not sure if it should be applied here too 🤷‍♀️

Reply:
Nope - in this case we test the SQL separately. He was even the one who suggested mocking teamManager:

https://github.com/PostHog/posthog/pull/12802#discussion_r1024541867